packages/horizontal-layout/src/vaadin-horizontal-layout-styles.js

```diff
@@ -28,3 +28,20 @@ export const horizontalLayoutStyles = css`
     gap: 1em;
   }
 `;
+
+// Layout improvements are part of a feature for Flow users where children that have been configured to use full size
+// using `HasSize.setSizeFull()` and others, get additional styles so that they effectively take the remaining space in
+// the layout, rather than explicitly use 100% width/height. The respective data attributes are set by Flow's `HasSize`
+// class.
+const enableLayoutImprovements = window.Vaadin.featureFlags.layoutImprovements;
+const layoutImprovementStyles = css`
+  ::slotted([data-width-full]) {
+    flex: 1;
+  }
+
+  ::slotted(vaadin-horizontal-layout[data-width-full]),
+  ::slotted(vaadin-vertical-layout[data-width-full]) {
+    min-width: 0;
+`;
```

Comment:
missing closing bracket:

```suggestion
  ::slotted(vaadin-vertical-layout[data-width-full]) {
    min-width: 0;
  } 
`;